File: Add Code Here/C++/tree/MyFirstApplication/app/src/main/java/com/example/myfirstapplication/SignupActivity.java
package com.example.myfirstapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.provider.ContactsContract;
import android.view.View;
import android.widget.EditText;

import com.example.myfirstapplication.ui.login.LoginActivity;

public class SignupActivity extends AppCompatActivity {

    EditText email;
    EditText password;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);
        email=(EditText)findViewById(R.id.editTextTextEmailAddress);
        password=(EditText)findViewById(R.id.editTextTextPassword);
        Intent mainCall=getIntent();
    }

    public void signUp(View view) {
        if(email.getText().toString().trim().equals("abc")&&(password.getText().toString().trim().equals("abc"))){
            Intent callLogin=new Intent(getApplicationContext(), LoginActivity.class);
            callLogin.putExtra("originalEmail",email.getText().toString().trim());
            callLogin.putExtra("originalPassword",password.getText().toString().trim());
            startActivity(callLogin);
        }
    }
}

File: Add Code Here/KOTLIN/Firebase Email and Password Authentication/SignUpActivity.kt
// package cn vary

import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import subrata.practice.firebase.android.authy.databinding.ActivitySignUpBinding

class SignUpActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySignUpBinding // used viewbinding
    private lateinit var firebaseAuth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySignUpBinding.inflate(layoutInflater)
        setContentView(binding.root)

        firebaseAuth = FirebaseAuth.getInstance()
        supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#FF3700B3")))

        binding.signUpBtn.setOnClickListener {
            val email = binding.signUpEmailEt.text.toString()
            val pass = binding.signUpPassEt.text.toString()

            if(!(email.isEmpty() && pass.isEmpty())) {
                firebaseAuth.createUserWithEmailAndPassword(email, pass).addOnCompleteListener {
                    if(it.isSuccessful) {
                        val intent = Intent(this, LogInActivity::class.java)
                        startActivity(intent)
                    } else {
                        Toast.makeText(this, it.exception.toString(), Toast.LENGTH_SHORT).show()
                    }
                }
            } else {
                Toast.makeText(this, "Empty fields are not allowed", Toast.LENGTH_SHORT).show()
            }
        }
    }
}